lib/net.c

```diff
@@ -47,32 +48,278 @@ find_num_numa_nodes(void)
 	return nb_numa_nodes;
 }
 
+int
+lua_init_iface(struct gatekeeper_if *iface, const char *iface_name,
+	const char **pci_addrs, uint8_t num_pci_addrs)
+{
+	uint8_t i, j;
+
+	iface->num_ports = num_pci_addrs;
+
+	iface->name = rte_malloc("iface_name", strlen(iface_name) + 1, 0);
+	if (iface->name == NULL) {
+		RTE_LOG(ERR, MALLOC, "%s: Out of memory for iface name\n",
+			__func__);
+		return -1;
+	}
+	strcpy(iface->name, iface_name);
+
+	iface->pci_addrs = rte_calloc("pci_addrs", num_pci_addrs,
+		sizeof(*pci_addrs), 0);
+	if (iface->pci_addrs == NULL) {
+		RTE_LOG(ERR, MALLOC, "%s: Out of memory for PCI array\n",
+			__func__);
+		goto name;
+	}
+
+	for (i = 0; i < num_pci_addrs; i++) {
+		iface->pci_addrs[i] = rte_malloc(NULL,
+			strlen(pci_addrs[i]) + 1, 0);
+		if (iface->pci_addrs[i] == NULL) {
+			RTE_LOG(ERR, MALLOC,
+				"%s: Out of memory for PCI address %s\n",
+				__func__, pci_addrs[i]);
+			for (j = 0; j < i; j++)
+				rte_free(iface->pci_addrs[j]);
+			rte_free(iface->pci_addrs);
```

Comment:
Please set `iface->pci_addrs = NULL` after `rte_free()`. See justification in a comment below.